test/plugins/variational_constraints/variational_constraints_engine_tests.jl

```diff
@@ -1115,3 +1200,90 @@ end
 end
+
+@testitem "Apply constraints to matrix variables" begin
+    import GraphPPL: getproperties, PluginsCollection, VariationalConstraintsPlugin, getextra, getcontext, with_plugins, create_model
+
+    include("../../testutils.jl")
+
+    using .TestUtils.ModelZoo
+
+    # Test for constraints applied to a model with matrix variables
+    c = @constraints begin
+        q(x, y) = q(x)q(y)
+    end
+    model = create_model(with_plugins(filled_matrix_model(), PluginsCollection(VariationalConstraintsPlugin(c))))
+
+    for node in filter(as_node(Normal), model)
+        @test getextra(model[node], :factorization_constraint_indices) == ([1], [2], [3])
+    end
+
+    @model function uneven_matrix()
+        local prec
+        local y
+        for i in 1:3
+            for j in 1:3
+                prec[i, j] ~ Gamma(1, 1)
+                y[i, j] ~ Normal(0, prec[i, j])
+            end
+        end
+        prec[2, 4] ~ Gamma(1, 1)
+        y[2, 4] ~ Normal(0, prec[2, 4])
+    end
+    constraints_1 = @constraints begin
+        q(prec, y) = q(prec)q(y)
+    end
+
+    model = create_model(with_plugins(uneven_matrix(), PluginsCollection(VariationalConstraintsPlugin(constraints_1))))
+    for node in filter(as_node(Normal), model)
+        @test getextra(model[node], :factorization_constraint_indices) == ([1], [2], [3])
+    end
+
+    constraints_2 = @constraints begin
+        q(prec[1], y) = q(prec[1])q(y)
+    end
+
+    model = create_model(with_plugins(uneven_matrix(), PluginsCollection(VariationalConstraintsPlugin(constraints_2))))
+    ctx = getcontext(model)
+    for node in filter(as_node(Normal), model)
+        if any(x -> x ∈ GraphPPL.neighbors(model, node), ctx[:prec][1])
+            @test getextra(model[node], :factorization_constraint_indices) == ([1], [2], [3])
+        else
+            @test getextra(model[node], :factorization_constraint_indices) == ([1, 3], [2])
+        end
+    end
+
+    constraints_3 = @constraints begin
+        q(prec[2], y) = q(prec[2])q(y)
+    end
+
+    model = create_model(with_plugins(uneven_matrix(), PluginsCollection(VariationalConstraintsPlugin(constraints_3))))
+    ctx = getcontext(model)
+    for node in filter(as_node(Normal), model)
+        if any(x -> x ∈ GraphPPL.neighbors(model, node), ctx[:prec][2])
+            @test getextra(model[node], :factorization_constraint_indices) == ([1], [2], [3])
+        else
+            @test getextra(model[node], :factorization_constraint_indices) == ([1, 3], [2])
+        end
+    end
+
+    constraints_4 = @constraints begin
+        q(prec[1, 3], y) = q(prec[1, 3])q(y)
+    end
+    model = create_model(with_plugins(uneven_matrix(), PluginsCollection(VariationalConstraintsPlugin(constraints_4))))
+    ctx = getcontext(model)
+    for node in filter(as_node(Normal), model)
+        if ctx[:prec][1, 3] ∈ GraphPPL.neighbors(model, node)
```

Comment:
Is this logic different from 
```
any(x -> x ∈ GraphPPL.neighbors(model, node), ctx[:prec][2])
```
which is used above for the same test?

Reply:
Because `prec[1,3]` is not a collection, but an individual `NodeLabel`. I can change it for consistency, but it shouldn't really matter